stubs/redis/redis/client.pyi

```diff
@@ -684,38 +684,297 @@ class PubSub:
+    def persist(self, name) -> Pipeline: ...
+    def pexpire(self, name, time) -> Pipeline: ...
+    def pexpireat(self, name, when) -> Pipeline: ...
+    def psetex(self, name, time_ms, value) -> Pipeline: ...
+    def pttl(self, name) -> Pipeline: ...
+    def randomkey(self) -> Pipeline: ...
+    def rename(self, src, dst) -> Pipeline: ...
+    def renamenx(self, src, dst) -> Pipeline: ...
+    def restore(self, name, ttl, value) -> Pipeline: ...
+    def set(
+        self,
+        name: _Key,
+        value: _Value,
+        ex: Union[None, int, timedelta] = ...,
+        px: Union[None, int, timedelta] = ...,
+        nx: bool = ...,
+        xx: bool = ...,
+        keepttl: bool = ...,
+    ) -> Pipeline: ...
+    def __setitem__(self, name, value) -> None: ...
+    def setbit(self, name: _Key, offset: int, value: int) -> Pipeline: ...
+    def setex(self, name: _Key, time: Union[int, timedelta], value: _Value) -> Pipeline: ...
+    def setnx(self, name, value) -> Pipeline: ...
+    def setrange(self, name, offset, value) -> Pipeline: ...
+    def strlen(self, name) -> Pipeline: ...
+    def substr(self, name, start, end=...) -> Pipeline: ...
+    def ttl(self, name: _Key) -> Pipeline: ...
+    def type(self, name) -> Pipeline: ...
+    def watch(self, *names: _Key) -> bool: ...
+    def unlink(self, *names: _Key) -> Pipeline: ...
+    def unwatch(self) -> bool: ...
+    def blpop(self, keys: Union[_Value, Iterable[_Value]], timeout: float = ...) -> Pipeline: ...
+    def brpop(self, keys: Union[_Value, Iterable[_Value]], timeout: float = ...) -> Pipeline: ...
+    def brpoplpush(self, src, dst, timeout=...) -> Pipeline: ...
+    def lindex(self, name: _Key, index: int) -> Pipeline: ...
+    def linsert(
+        self, name: _Key, where: Literal["BEFORE", "AFTER", "before", "after"], refvalue: _Value, value: _Value
+    ) -> Pipeline: ...
+    def llen(self, name: _Key) -> Pipeline: ...
+    def lpop(self, name) -> Pipeline: ...
+    def lpush(self, name: _Value, *values: _Value) -> Pipeline: ...
+    def lpushx(self, name, value) -> Pipeline: ...
+    def lrange(self, name: _Key, start: int, end: int) -> Pipeline: ...
+    def lrem(self, name: _Key, count: int, value: _Value) -> Pipeline: ...
+    def lset(self, name: _Key, index: int, value: _Value) -> Pipeline: ...
+    def ltrim(self, name: _Key, start: int, end: int) -> Pipeline: ...
+    def rpop(self, name) -> Pipeline: ...
+    def rpoplpush(self, src, dst) -> Pipeline: ...
+    def rpush(self, name: _Value, *values: _Value) -> Pipeline: ...
+    def rpushx(self, name, value) -> Pipeline: ...
+    def sort(
+        self,
+        name: _Key,
+        start: Optional[int] = ...,
+        num: Optional[int] = ...,
+        by: Optional[_Key] = ...,
+        get: Optional[Union[_Key, Sequence[_Key]]] = ...,
+        desc: bool = ...,
+        alpha: bool = ...,
+        store: Optional[_Key] = ...,
+        groups: bool = ...,
+    ) -> Pipeline: ...
```

Comment:
`Redis().sort()` is overloaded to provide an accurate return type, but for `Pipeline` the return type is always `Pipeline`, so I removed the overloads.

https://github.com/python/typeshed/blob/e2fd852952ff807c60819f8840e8b23664cc7593/stubs/redis/redis/client.pyi#L495-L534